src/mlpack/core/tree/cover_tree/cover_tree_impl.hpp

```diff
@@ -587,6 +658,65 @@ CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::CoverTree(
   other.metric = NULL;
 }
 
+// Move Assignment.
+template<
+    typename MetricType,
+    typename StatisticType,
+    typename MatType,
+    typename RootPointPolicy
+>
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>&
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::
+operator=(CoverTree&& other)
+{
+  if (this == &other)
+    return *this;
+
+  // Freeing memory that will not be used anymore.
+  if (localDataset)
+    delete dataset;
+
+  if (localMetric)
+    delete metric;
+
+  for (size_t i = 0; i < children.size(); ++i)
+    delete children[i];
+  children.clear();
```

Comment:
It doesn't hurt anything to leave it here, but the `clear()` isn't necessary for the move constructor, since `std::move(other.children)` will set the right size.

Reply:
Done 🙂